### PR TITLE
Preserve trivia better when converting if to switch.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/21109
Fixes https://github.com/dotnet/roslyn/issues/21101
Fixes https://github.com/dotnet/roslyn/issues/21103
Fixes https://github.com/dotnet/roslyn/issues/20989
Fixes https://github.com/dotnet/roslyn/issues/21108
Fixes https://github.com/dotnet/roslyn/issues/19733

### CODE REVIEW

File: src/EditorFeatures/VisualBasicTest/CodeActions/ConvertIfToSwitch/ConvertIfToSwitchTests.vb [hunk 1]
```diff
@@ -360,5 +360,39 @@ End Class",
...
+    End Sub
+End Class",
+"Class C
+    Sub M(i As Integer)
+#if true
+        Console.WriteLine()
+#end if
+
+        Select i
+            Case 1, 2, 3             M(0)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/CodeActions/ConvertIfToSwitch/ConvertIfToSwitchTests.vb]: Note: this is broken.  but tracked with another bug.

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
retest windows_release_vs-integration_prtest please

**Comment by dpoeschl** [comment]
VS test failures are being investigated.

**Pull request merged**